cmd/promtool/main.go

```diff
@@ -171,16 +176,96 @@ func checkRules(t cli.Term, filename string) (int, error) {
 		return 0, fmt.Errorf("is a directory")
 	}
 
+	rgs, errs := rulefmt.ParseFile(filename)
+	if errs != nil {
+		return 0, tsdb.MultiError(errs)
+	}
+
+	numRules := 0
+	for _, rg := range rgs.Groups {
+		numRules += len(rg.Rules)
+	}
+
+	return numRules, nil
+}
+
+// UpdateRulesCmd updates the rule files.
+func UpdateRulesCmd(t cli.Term, args ...string) int {
+	if len(args) == 0 {
+		t.Infof("usage: promtool update-rules <files>")
+		return 2
+	}
+	failed := false
+
+	for _, arg := range args {
+		if err := updateRules(t, arg); err != nil {
+			t.Errorf("  FAILED: %s", err)
+			failed = true
+		}
+	}
+
+	if failed {
+		return 1
+	}
+	return 0
+}
+
+func updateRules(t cli.Term, filename string) error {
+	t.Infof("Updating %s", filename)
+
+	if stat, err := os.Stat(filename); err != nil {
+		return fmt.Errorf("cannot get file info")
+	} else if stat.IsDir() {
+		return fmt.Errorf("is a directory")
+	}
+
 	content, err := ioutil.ReadFile(filename)
 	if err != nil {
-		return 0, err
+		return err
 	}
 
 	rules, err := promql.ParseStmts(string(content))
 	if err != nil {
-		return 0, err
+		return err
+	}
+
+	yamlRG := &rulefmt.RuleGroups{
+		Version: 1,
+		Groups: []rulefmt.RuleGroup{{
+			Name: filename,
+		}},
+	}
+
+	yamlRules := make([]rulefmt.Rule, 0, len(rules))
+
+	for _, rule := range rules {
+		switch r := rule.(type) {
+		case *promql.AlertStmt:
+			yamlRules = append(yamlRules, rulefmt.Rule{
+				Alert:       r.Name,
+				Expr:        r.Expr.String(),
+				For:         model.Duration(r.Duration),
+				Labels:      r.Labels.Map(),
+				Annotations: r.Annotations.Map(),
+			})
+		case *promql.RecordStmt:
+			yamlRules = append(yamlRules, rulefmt.Rule{
+				Record: r.Name,
+				Expr:   r.Expr.String(),
+				Labels: r.Labels.Map(),
+			})
+		default:
+			panic("unknown statement type")
+		}
+	}
+
+	yamlRG.Groups[0].Rules = yamlRules
+	y, err := yaml.Marshal(yamlRG)
```

Comment:
Because we are now using the plain YAML again this will generate rule files with basically random key ordering right?

@brian-brazil if yes, this is pretty bad UX just because we don't want to use a 100 LOC wrapper lib that's vendored already anyway.